test/metabase/db/custom_migrations_test.clj

```diff
@@ -1680,3 +1692,42 @@
               (is (= #{(#'api.database-test/sync-and-analyze-trigger-name db)}
                      (#'task.sync-databases-test/query-all-db-sync-triggers-name db)))
               (is (nil? (:cache_field_values_schedule db))))))))))
+
+(defn scheduler-job-keys
+  []
+  (->> (task/scheduler-info)
+       :jobs
+       (map :key)
+       set))
+
+(deftest delete-send-pulse-job-on-migrate-down-test
+  (impl/test-migrations ["v50.2024-04-25T01:04:06" "v50.2024-04-25T01:04:08"] [migrate!]
+    (migrate!)
+    (pulse-channel-test/with-send-pulse-setup!
+      (let [user-id  (:id (new-instance-with-default :core_user))
+            pulse-id (:id (new-instance-with-default :pulse {:creator_id user-id}))
+            pc       (new-instance-with-default :pulse_channel {:pulse_id pulse-id})]
```

Comment:
Why use `new-instance-with-default` here? I don't think the DRY principle applies here. We don't want to couple migrations to each other in case the underlying schema or defaults for the tables change over time.

Reply:
that's the idea of `new-instance-with-default` being defined in this namespace, otherwise we could use `t2.with-temp/with-temp-defaults`.

`new-instance-with-default` should only have fields that are less likely to be changed, like core_user.email, created_at ...

maybe schedule_type and schedule_hour are not the best candidates, as we plan to remove them. Nevertheless  I think some amount of DRY is valuable here as these migrations are often very long and take some amount of setup to test.

Reply:
> some amount of DRY is valuable here

agreed, DRYing up things to keep code shorter is good, except for when it can cause issues like here

> new-instance-with-default should only have fields that are less likely to be changed

true, it was maybe okay for users. but it becomes a problem when we use it for things that are more likely to change, so I think it should be discouraged. Not a strong opinion and I see the upsides too. Just thought I might point it out